python/bifrost/units.py

```diff
@@ -31,13 +31,16 @@
 ureg = pint.UnitRegistry()
 
 def convert_units(value, old_units, new_units):
-	old_quantity = value * ureg.parse_expression(old_units)
-	try:
-		new_quantity = old_quantity.to(new_units)
-	except pint.DimensionalityError:
-		raise ValueError("Cannot convert units %s to %s" %
+    old_quantity = value * ureg.parse_expression(old_units)
+    try:
+        new_quantity = old_quantity.to(new_units)
+    except pint.DimensionalityError:
+        raise ValueError("Cannot convert units %s to %s" %
 		                 (old_units, new_units))
-	return new_quantity.magnitude
+    if value >= 0:
+        return new_quantity.magnitude
+    else:
+        return -1 * new_quantity.magnitude
```

Comment:
To confirm, pint 'overloaded' the term 'magnitude' to mean 'value, not magnitude':
```python
In [1]: import pint

In [2]: ureg = pint.UnitRegistry()

In [3]: a = 1.0 * ureg.meters

In [4]: a
Out[4]: <Quantity(1.0, 'meter')>

In [5]: a.magnitude
Out[5]: 1.0

In [6]: a = -1.0 * ureg.meters

In [7]: a.magnitude
Out[7]: -1.0
```